skiko/src/nativeJsMain/cpp/Typeface.cc

```diff
@@ -20,58 +20,46 @@ SKIKO_EXPORT KBoolean org_jetbrains_skia_Typeface__1nIsFixedPitch
     return instance->isFixedPitch();
 }
 
-
-SKIKO_EXPORT KInteropPointerArray org_jetbrains_skia_Typeface__1nGetVariations
+SKIKO_EXPORT KInt org_jetbrains_skia_Typeface__1nGetVariationsCount
   (KNativePointer ptr) {
-    TODO("implement org_jetbrains_skia_Typeface__1nGetVariations");
+    SkTypeface* instance = reinterpret_cast<SkTypeface*>(ptr);
+    return instance->getVariationDesignPosition(nullptr, 0);
 }
-     
-#if 0 
-SKIKO_EXPORT KInteropPointerArray org_jetbrains_skia_Typeface__1nGetVariations
-  (KNativePointer ptr) {
-    SkTypeface* instance = reinterpret_cast<SkTypeface*>((ptr));
-    int count = instance->getVariationDesignPosition(nullptr, 0);
+
+SKIKO_EXPORT void org_jetbrains_skia_Typeface__1nGetVariations
+  (KNativePointer ptr, KInt* res, KInt count) {
+    SkTypeface* instance = reinterpret_cast<SkTypeface*>(ptr);
     if (count > 0) {
         std::vector<SkFontArguments::VariationPosition::Coordinate> coords(count);
         instance->getVariationDesignPosition(coords.data(), count);
-        KInteropPointerArray res = env->NewObjectArray(count, skija::FontVariation::cls, nullptr);
         for (int i=0; i < count; ++i) {
-            KInteropPointer var = env->NewObject(skija::FontVariation::cls, skija::FontVariation::ctor, coords[i].axis, coords[i].value);
-            env->SetObjectArrayElement(res, i, var);
+             int r[2] = {static_cast<int>(coords[i].axis), rawBits(coords[i].value)};
+             memcpy(res + 2 * i, r, sizeof r);
```

Comment:
IMHO `res[2*i] = static_cast<KInt>(coords[i].axis); res[2*i + 1] = rawBits(coords[i].value)` is much simpler

Reply:
+1

Reply:
yeah, that totally make sense - this is a trace of jni-influence code, pushed changes